### ISSUE TITLE
No way to Manipulate DataAPI in Aurora Serverless V2

### ISSUE THREAD

**Issue by TiagoJVO** [terraform-aws-modules/terraform-aws-rds-aurora#464]
## Description

TLDR:
In Aurora Serverless V2 manipulating `enable_http_endpoint ` only applies on resource creation, subsequent changes of the property will lead to every reconciliation to try to apply changes due to interpreting a difference between the resource and the definition, but leading to no change (as in, keeps reconciling every 12h applying no change).

 
If you set `enable_http_endpoint` to `true` at creation time, it is correctly set via `EnableHttpEndpoint` mentioned on the following API documentation page:
[AWS CreateDBCluster Operation](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html)
 
When it comes to modification of said option, I see it is implemented in the code as:
[terraform-provider-aws code](https://github.com/hashicorp/terraform-provider-aws/blob/3724def77810f619392ecb79fd3e40915c0b0214/internal/service/rds/cluster.go#L1455)
 
The issue is, the `EnableHttpEndpoint` passed to `ModifyDBCluster` action works only for Aurora Serverless V1. This means if the cluster has been created with HTTP Data API disabled, enabling it via this API parameter will work only for v1. 
V2, which we use, has a completely separate action, mentioned here:
[AWS EnableHttpEndpoint Operation](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_EnableHttpEndpoint.html)
 
This does not seem to be supported by the provider, and the path seems to be forking the code depending on the version, and for v2 calling the separate operation instead of modifying the cluster.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.
Did check and there is a closed issue on this topic, that was closed due to inactivity: [Related Issue](https://www.infoq.com/news/2024/01/aurora-serverless-v1-retirement/#:~:text=We%20are%20reaching%20out%20to,Serverless%20version%201%20(v1).)

## Versions

- Module version [Required]:
v7+
- Terraform version / Provider version(s):
N/A

## Reproduction Code [Required]
```
variable "region" {
  default = "us-west-2"
}

module "my_db" {
  source      = "terraform-aws-modules/rds-aurora/aws"
  version     = "~> 7.1"
  engine      = "aurora-postgresql"
  engine_mode = "provisioned"
  engine_version = 13.12
  instance_class = "db.serverless"
  # https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.Aurora_Fea_Regions_DB-eng.Feature.Data_API.html#Concepts.Aurora_Fea_Regions_DB-eng.Feature.Data_API.apg
  enable_http_endpoint = true
  name = "case-number-157917"
  subnets = [ "subnet-06259998c0dab749d", "subnet-02856b15739009107" ]
  apply_immediately = true
}
```

1. Apply the above TF
2. Verify the DB is created with DataAPI enabled
3. change `enable_http_endpoint` to `false`
4. apply TF
5. verify DataAPI is not disabled
6. every subsequent reconcile will find a difference and try to apply the modification, but no modification is done (consuming resources)

## Expected behavior

`enable_http_endpoint` to manipulate DataAPI toggle for Aurora Serverless v2

## Actual behavior

`enable_http_endpoint` only works on Creation time for Aurora Serverless v2

## Additional context

This is specially important to be implemented now as AWS Announced the [retirement of Aurora Serverless V1 by December 2024](https://www.infoq.com/news/2024/01/aurora-serverless-v1-retirement/#:~:text=We%20are%20reaching%20out%20to,Serverless%20version%201%20(v1).), and all databases will be migrated to v2 (if not migrated by ourselves).
